protocols/v2/framing-sv2/src/framing.rs

```diff
@@ -1,57 +1,56 @@
-#![allow(dead_code)]
-use crate::{
-    header::{Header, NOISE_HEADER_LEN_OFFSET, NOISE_HEADER_SIZE},
-    Error,
-};
+use crate::{header::Header, Error};
 use alloc::vec::Vec;
 use binary_sv2::{to_writer, GetSize, Serialize};
 use core::convert::TryFrom;
 
-const NOISE_MAX_LEN: usize = const_sv2::NOISE_FRAME_MAX_SIZE;
-
 #[cfg(not(feature = "with_buffer_pool"))]
 type Slice = Vec<u8>;
 
 #[cfg(feature = "with_buffer_pool")]
 type Slice = buffer_sv2::Slice;
 
-impl<A, B> Sv2Frame<A, B> {
-    /// Maps a `Sv2Frame<A, B>` to `Sv2Frame<C, B>` by applying `fun`,
-    /// which is assumed to be a closure that converts `A` to `C`
-    pub fn map<C>(self, fun: fn(A) -> C) -> Sv2Frame<C, B> {
-        let serialized = self.serialized;
-        let header = self.header;
-        let payload = self.payload.map(fun);
-        Sv2Frame {
-            header,
-            payload,
-            serialized,
+/// A wrapper to be used in a context we need a generic reference to a frame
+/// but it doesn't matter which kind of frame it is (`Sv2Frame` or `HandShakeFrame`)
+#[derive(Debug)]
+pub enum Frame<T, B>
+where
+    T: Serialize + GetSize,
```

Comment:
Best practice is to constrain generics only where you need it.

Reply:
In the impl block where you have function that need the generics to be constrained

Reply:
Having naked generics makes it really hard to read the enum

Reply:
I don't get it. Why is harder?

Reply:
Maybe you want to use InnerType and Buffer instead of T and B ?

Reply:
Btw we shouldn't restrain generics if there is no need

Reply:
previously it was only `Frame<T,B>{ HandShake(HandShakeFrame), Sv2(Sv2Frame<T,B>)` without any info about T and B

Reply:
I went trough the code everything look very good to me.

Reply:
what is the outcome here?

@Fi3 do you still feel need we shouldn't be constraining generics here? or looking at the rest of the PR change your mind?

Reply:
constrain generics where not needed is bad, so my opinion is the same. If you are concerned about readability you can use Frame<InnerType,Buffer> instead of Frame<T,B>